src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -2971,14 +2971,6 @@ namespace X{
             diag::Diagnostics& diagnostics) {                                              \
         ASRUtils::require_impl(x.n_args == 2, "Intrinsic function `"#X"` accepts           \
             exactly 2 arguments", x.base.base.loc, diagnostics);                           \
-                                                                                           \
```

Comment:
cc @Thirumalai-Shaktivel 
For now I've removed this as of possibility of what we had discussed but we might want to keep track of this and decide what needs to be done here !

Reply:
Do you want to check for the `CPtr` type?

Reply:
Actually what I realized is, this might not be an issue just yet .... For a program like the following where we are dealing with everything through simple assignment and print statements , it might be an issue atleast for implementations in this PR.
```
from lpython import S
from sympy import Symbol, diff, sin, pi

def main0():
    x: S = Symbol("x")
    y: S = sin(x)
    z: S = x + y
    a: S = pi
    b: S = a**z
    print(z)
    print(b)
```
Though I know of somecases where we might encounter this issue and we would tackle it in the upcoming PRs.